### PR TITLE
disable tests who require en culture, if current culture is not en

### LINKED ISSUE #1118: Tests broken on german machine because warning is given in german
Our tests are not independent of the language settings of the developer machine.

image

#### Repro steps

Use german machine and run `build.cmd all`

#### Actual behaviour

Build is broken with the following changes:

image

image

image

image

### PR DESCRIPTION
**Pull request by enricosada**
fix #1118

tests who require `en` culture are disabled, unless culture is `en`

### CODE REVIEW

**Comment by forki** [comment]
Can we make it require en-US?

File: tests/fsharp/core/tests_core.fs [hunk 1]
```diff
@@ -676,6 +682,8 @@ module Printing =
     let printing flag diffFileOut expectedFileOut diffFileErr expectedFileErr = check (attempt {
         let { Directory = dir; Config = cfg } = testContext ()
 
+        do! requireENCulture ()
```
Comment (forki) [inline tests/fsharp/core/tests_core.fs]: Does this skip all tests?

**Comment by enricosada** [comment]
@forki i *think* @dsyme is en-GB, so i dont want to disable test for him, `en` is enough

**Comment by forki** [comment]
yeah but I also want to run tests ;-)

File: tests/fsharp/core/tests_core.fs (hunk 1 above)
Comment (enricosada) [inline tests/fsharp/core/tests_core.fs]: yes, it's temporary, until `--preferreduilang` is merged. After that i reenable tests,
meanwhile the appveyor/jenkins run tests anyway, so no issues.

Comment (enricosada) [inline tests/fsharp/core/tests_core.fs]: it disable `negTest` only, not all

File: tests/fsharp/typeProviders/tests_typeProviders.fs [hunk 2]
```diff
@@ -21,6 +21,13 @@ let requireVSUltimate cfg = attempt {
             // )
     }
 
+let requireENCulture () = attempt {
```
Comment (dsyme) [inline tests/fsharp/typeProviders/tests_typeProviders.fs]: can we share this code with the above?

**Comment by enricosada** [comment]
you are `en-What`?

**Comment by forki** [comment]
I'm de-DE ;-) I want to run all tests, but not the 4 that only work on en-US

**Comment by enricosada** [comment]
@forki that's my pr :smile: , i disable the 5 tests ( 4 `core/printing` and `netTest`) who cannot run without en culture (i have `it-IT` )

**Comment by forki** [comment]
very good!

File: tests/fsharp/typeProviders/tests_typeProviders.fs (hunk 2 above)
Comment (enricosada) [inline tests/fsharp/typeProviders/tests_typeProviders.fs]: yes, i was a bit lazy, thx @dsyme

**Comment by enricosada** [comment]
ok refactored out the functions

**Comment by enricosada** [comment]
that's ok for review/merge /cc @KevinRansom @dsyme

**Comment by forki** [comment]
image

what about that one?

**Pull request merged**